DOCS-5894 filter conditions (#105)
mulesoft/docs-api-community-manager: modules/ROOT/pages/acm-lightning-components.adoc

Card Border Radius:: Specifies the card border radius. A larger radius makes the cards look more rounded.
Card Shadow:: Specifies the shadow beneath each card.
Message If Empty:: Shows a message if the carousel has no APIs. Default is `There are no APIs available right now.`

> Filter Condition (Advanced):: Causes the carousel to show only APIs that match the given search.

Reviewer: please indicate that 'the syntax for this field is described in the "Filter conditions" section below'

Author: Fixed.

API Details Button Label:: Enables you to set the API Details button label. Default is *API Details*.
Learn More Button Label:: Enables you to set the Learn More button label. Default is *Learn More*.

=== Filter Conditions

The API Carousel component parameter *Filter Condition (Advanced)* causes the carousel to filter out the APIs that do not match the condition and show only the APIs that match. A condition is a standard search in Salesforce Object Query Language (SOQL).

> You can search these fields:

Reviewer: Double check the package prefix - current one is the DEV, for publication we need to publish with the PRODUCTION package name.

Author: Thank you! Messaged Fernando about this and he confirmed these field names. `acm_pkg__Icon__c` `acm_pkg__Name__c` `acm_pkg__NavigateUrl__c` `acm_pkg__VersionGroup__c`

> * `acm_pkg_dev__Icon__c`

Reviewer: there seems to be a problem when markdown renders this text, because of the "_" - when I see it rendered in github, I read "acm_pkg_devIconc". Please double-check if/how to escape _ and __ to make sure the field names are displayed correctly.

Author: You're right. Sorry about that! Fixed.

* `acm_pkg_dev__Name__c`
* `acm_pkg_dev__NavigateUrl__c`
* `acm_pkg_dev__VersionGroup__c`

> These examples demonstrate filter conditions.

Reviewer: please add two more examples:
1. illustrating OR:
API NAMe is "My API A" or "My API B"...

2. illustrating AND:
API Name is "My API" and VersionGroup is "v1"

Reviewer: observe that for OR/AND usage, the whole condition must be enclosed in () - for example:

(acm_pkg_dev__Name__c LIKE 'Product%'  OR acm_pkg_dev__Name__c LIKE 'Shipping%' OR acm_pkg_dev__Name__c LIKE 'Order%')

Author: Great info! Added.

Card Border Radius:: Specifies the card border radius. A larger radius makes the cards look more rounded.
Card Shadow:: Specifies the shadow beneath each card.
Message If Empty:: Shows a message if the carousel has no APIs. Default is `There are no APIs available right now.`

> Filter Condition (Advanced):: Causes the carousel to show only APIs that match the given search condition, which uses the syntax described in the following section.

Reviewer 2: Beware ever referring to another section, because things move around and change. Better to name the title or refer to it generally: "...described in the filter conditions documentation."

Author: Good point. Updated this to name the title and link to the section.

API Details Button Label:: Enables you to set the API Details button label. Default is *API Details*.
Learn More Button Label:: Enables you to set the Learn More button label. Default is *Learn More*.

=== Filter Conditions

> The API Carousel component parameter *Filter Condition (Advanced)* causes the carousel to filter out the APIs that do not match the condition and show only the APIs that match. A condition is a standard search in Salesforce Object Query Language (SOQL).

Reviewer 2: ```suggestion
The API Carousel component parameter *Filter Condition (Advanced)* causes the carousel to show only the APIs that meet the condition. A condition is a standard search in Salesforce Object Query Language (SOQL).
```

> You can search these fields.

Reviewer 2: ```suggestion
You can search these fields:
```
GLOBALLY introduce lists, steps, syntax, and other graphic displays with a sentence or fragment that ends with a colon.

Author: Sorry about that. I mistakenly thought the style guide said the opposite. I'll remember this and introduce these with a colon from now on.

* `+acm_pkg__Icon__c+`
* `+acm_pkg__Name__c+`
* `+acm_pkg__NavigateUrl__c+`
* `+acm_pkg__VersionGroup__c+`

> These examples demonstrate filter conditions.

Reviewer 2: ditto

Author: Fixed.

* The API name contains the text `My API`: `+acm_pkg__Name__c LIKE '%My API%'+`
* The API name is `API Alpha` or `API Bravo`: `+(acm_pkg__Name__c = 'API Alpha' OR acm_pkg__Name__c = 'API Bravo')+`
* The API name is `My API` and the API version group is `v1`: `+(acm_pkg__Name__c = 'My API' AND acm_pkg__VersionGroup__c = 'v1')+`
* The API name begins with `Product`, `Shipping`, or `Order`: `+(acm_pkg__Name__c LIKE 'Product%' OR acm_pkg__Name__c LIKE 'Shipping%' OR acm_pkg__Name__c LIKE 'Order%')+`
* The API has an icon: `+acm_pkg__Icon__c != ''+`
* The API has a Learn More button that links to a custom page with more information: `+acm_pkg__NavigateUrl__c != ''+`
* The API version group is exactly equal to `v2`: `+acm_pkg__VersionGroup__c = 'v2'+`
* The API version group is later than `v1`: `+acm_pkg__VersionGroup__c > 'v1'+`
* The API version group is earlier than `v1`: `+acm_pkg__VersionGroup__c < 'v1'+`
* The API version group is later than or equal to `v2`: `+acm_pkg__VersionGroup__c >= 'v2'+`
* The API version group is earlier than or equal to `v2`: `+acm_pkg__VersionGroup__c <= 'v2'+`

Queries using `OR` or `AND` must enclose the whole condition in parentheses.

> ACM compares version groups using normal string comparison.

Reviewer 2: Does your audience know what normal string comparison is?

Author: Yes, these are developers and will be used to string comparison, and unsurprised that ['3', 'V2', 'v1', 'zzz'] is considered to be sorted from lowest to highest.
